jakarta-eclipse/src/org/jakarta/jdt/ServletDiagnosticsCollecter.java

```diff
@@ -26,28 +26,93 @@ public void collectDiagnostics(ICompilationUnit unit, List<Diagnostic> diagnosti
 				alltypes = unit.getAllTypes();
 				for (IType type : alltypes) {
 					allAnnotations = type.getAnnotations();
+					
+					
 					boolean isWebServletAnnotated = false;
 					boolean isHttpServletExtended = false;
+					
+					boolean isWebFilterAnnotated = false;
+					boolean isFilterImplemented = false;
+					
+					boolean isWebListenerAnnotated = false;
+					boolean isWebListenerInterfaceImplemented = false;
+					
 					for (IAnnotation annotation : allAnnotations) {
 						if (annotation.getElementName() == "WebServlet") {
 //							System.out.println("--Annotation name: " + annotation.getElementName());
 							isWebServletAnnotated = true;
 						}
+						if (annotation.getElementName() == "WebFilter") {
+							isWebFilterAnnotated = true;
+						}
+						if (annotation.getElementName() == "WebListener") {
+							isWebListenerAnnotated = true;
+						}
 					}
 
 					String typeExtension = type.getSuperclassName();
 //					System.out.println("--extension name: " + type.getSuperclassName());
 					if ((typeExtension != null) && typeExtension.compareTo("HttpServlet") == 0) {
 						isHttpServletExtended = true;
 					}
+					
+					String[] implementedInterfaces = type.getSuperInterfaceNames();
+					
+					for(String in: implementedInterfaces) {
+						if (in.compareTo("Filter") == 0) {
+							isFilterImplemented = true;
+						}
+						
+						if (in.compareTo("Filter") == 0) {
+							isFilterImplemented = true;
+						}
+						
+						if (in.compareTo("ServletContextListener") == 0) {
+							isWebListenerInterfaceImplemented = true;
+						}
+						if (in.compareTo("ServletContextAttributeListener") == 0) {
+							isWebListenerInterfaceImplemented = true;
+						}
+						if (in.compareTo("ServletRequestListener") == 0) {
+							isWebListenerInterfaceImplemented = true;
+						}
+						if (in.compareTo("ServletRequestAttributeListener") == 0) {
+							isWebListenerInterfaceImplemented = true;
+						}
+						if (in.compareTo("HttpSessionAttributeListener") == 0) {
+							isWebListenerInterfaceImplemented = true;
+						}
+						if (in.compareTo("HttpSessionAttributeListener") == 0) {
+							isWebListenerInterfaceImplemented = true;
+						}
+						if (in.compareTo("HttpSessionIdListener") == 0) {
+							isWebListenerInterfaceImplemented = true;
+						}
```

Comment:
Since this is looking for equality in string values, you could just write like `if (in.equals("Filter"))`